src/test/kotlin/com/igorwojda/string/ispalindrome/tolerant/challenge.kt

```diff
@@ -4,7 +4,11 @@ import org.amshove.kluent.shouldBeEqualTo
 import org.junit.jupiter.api.Test
 
 private fun isTolerantPalindrome(str: String, characterRemoved: Boolean = false): Boolean {
-
+    val (lettersOrDigits, nonLettersOrDigits) = str.partition { c -> c.isLetterOrDigit() }
+    val (letters, digits) = lettersOrDigits.partition { c -> c.isLetter() }
+    return if (nonLettersOrDigits.length > 1 || (letters.length > 1 && digits.length > 1)) false
```

Comment:
inconsistent formatting. Please use curly braces in if-else statements even if they are single line expressions. It makes it easier to read & debug.